executor/join.go

```diff
@@ -70,6 +69,8 @@ type HashJoinExec struct {
 
 	// closeCh add a lock for closing executor.
 	closeCh      chan struct{}
+	worker       util.WaitGroupWrapper
+	waiter       util.WaitGroupWrapper
```

Comment:
`waitJoinWorkersAndCloseResultChan` will call `worker.Wait()`, but it is running in another goroutine. So `waiter` is to prevent leak of `waitJoinWorkersAndCloseResultChan`.